admin/category_product_listing.php

```diff
@@ -462,32 +462,32 @@
             ?>
           </div>
           <div class="col-md-4">
-          <?php require DIR_WS_MODULES . 'search_box.php'; ?>
-            <?php echo zen_draw_form('goto', FILENAME_CATEGORY_PRODUCT_LISTING, '', 'get', 'class="form-horizontal"'); ?>
-            <?php echo zen_hide_session_id(); ?>
+          <?php require DIR_WS_MODULES . 'search_box.php' ?>
```

Comment:
Missing semicolon.

Reply:
steve,
in our discussions over the short echo tag, the question came up over the semicolon prior to the PHP closing tag.  there was a consensus to not have it.

this formatting style continues in this require statement.

perhaps one can get their AI bot to only remove the closing tag of short echo statements and not for other php statements, but my AI bot is not well trained.

[it is not required.](https://www.php.net/manual/en/language.basic-syntax.instruction-separation.php#:~:text=As%20in%20C%20or%20Perl,line%20of%20a%20PHP%20block.)

Reply:
I thought that we were omitting the semicolon only where the echo short tag was being used.

Reply:
The semi colon is allowed but not required.   No need for argument.

Also remember that a syntax error will be found by automatic tests, so never any need to worry about that.

Reply:
>i have better things to be faffing on.

As does everyone.

Perhaps you should put this time-waster in his place too:
https://github.com/zencart/zencart/commit/136e5964eebc7d466afff84e78d54f134e99ff97

Reply:
perhaps you should have properly directed your comment to whom you wanted it to go instead of posting it on this PR.

Reply:
It's true: We all have much more important things to spend our time on other than bickering about formatting standards.

Given that the major PHP projects all use the closing semicolon on pretty much all non-echo statements, and it limits developer error, Zen Cart also uses the closing semicolon on all statements except for "echo" statements.

|statement|standard|
|-----|-----|
|`<=`|closing semicolon not needed|
|`<?php echo`|closing semicolon not needed|
|`<?php`|closing semicolon required|

Coding standards documentation has been updated accordingly.